### PR TITLE
Enable DirectoryServices packages

### PR DESCRIPTION
**Pull request by ericstj**
/cc @weshaggard @jay98014 

A couple problems I dealt with:
1. These have the same assembly name as desktop, a higher version than desktop, and fewer types.  I dialed the version back to match desktop, though they still have fewer types.  You won't be able to add anything to these until you can support everything that desktop supported.  Alternatively give them different names than the desktop assemblies.
2. They are signing with the open key rather than the one used by desktop.
3. They don't support all runtimes.  I added a platform-not-supported assembly for non-windows runtimes.

### CODE REVIEW

File: pkg/descriptions.json [hunk 1]
```diff
@@ -569,6 +569,32 @@
...
+      "System.DirectoryServices.ActiveDirectory.DirectoryContext",
+      "System.DirectoryServices.ActiveDirectory.DirectoryServer",
+      "System.DirectoryServices.ActiveDirectory.Domain",
+      "System.DirectoryServices.ActiveDirectory.DomainController"
+    ]
+  },
+  {
+    "Name": "System.DirectoryServices.AccountManagement",
+    "Description": "Provides uniform access and manipulation of user, computer, and group security principals across the multiple principal stores: Active Directory Domain Services (AD DS), Active Directory Lightweight Directory Services (AD LDS), and Machine SAM (MSAM).",
+    "CommonTypes": [
```
Comment (ericstj) [inline pkg/descriptions.json]: Need a good list of entry-point types for this.

File: pkg/descriptions.json [hunk 2]
```diff
@@ -569,6 +569,32 @@
...
+    ]
+  },
+  {
+    "Name": "System.DirectoryServices.Protocols",
+    "Description": "Provides the methods defined in the Lightweight Directory Access Protocol (LDAP) version 3 (V3) and Directory Services Markup Language (DSML) version 2.0 (V2) standards.",
+    "CommonTypes": [
```
Comment (ericstj) [inline pkg/descriptions.json]: Need a good list of entry-point types for this.

File: src/System.DirectoryServices.Protocols/src/FxCopBaseline.AnyOS.cs [hunk 3]
```diff
@@ -0,0 +1 @@
+[assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Security", "CA1821", Justification = "Finalizer has implementation in Windows version.")]
```
Comment (ericstj) [inline src/System.DirectoryServices.Protocols/src/FxCopBaseline.AnyOS.cs]: Wasn't sure if I should do this or remove the finalizers from the reference assembly.  /cc @terrajobst

File: src/System.DirectoryServices.AccountManagement/dir.props [hunk 4]
```diff
@@ -2,7 +2,7 @@
 <Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <Import Project="..\dir.props" />
   <PropertyGroup>
-    <AssemblyVersion>4.1.1.0</AssemblyVersion>
+    <AssemblyVersion>4.0.0.0</AssemblyVersion>
     <AssemblyKey>Open</AssemblyKey>
```
Comment (weshaggard) [inline src/System.DirectoryServices.AccountManagement/dir.props]: These need to match the desktop key if you want to keep the same assembly name.

**Comment by weshaggard** [comment]
I don't think it is a good idea to keep the same assembly identity here as it is going to limit our options for evolving these.

File: src/System.DirectoryServices.AccountManagement/dir.props (hunk 4 above)
Comment (ericstj) [inline src/System.DirectoryServices.AccountManagement/dir.props]: Yeah, that's what I already pointed out.

**Comment by ericstj** [comment]
> I don't think it is a good idea to keep the same assembly identity here as it is going to limit our options for evolving these.
I agree.

**Comment by ericstj** [comment]
@weshaggard this was more a demonstration / example.  Do you want to merge it or close it and file an issue for the folks who own these libraries to fix the design before we package them?

**Comment by weshaggard** [comment]
>@weshaggard this was more a demonstration / example. Do you want to merge it or close it and file an issue for the folks who own these libraries to fix the design before we package them?

I would prefer we solve the issues before we enable the package builds. I'm fine with you merging in the changes beyond the package configuration.props so that the packages are still disabled.

**Comment by ericstj** [comment]
How about I explicitly disable Pack?  That way we still build the pkgproj and they don't get stale, we just don't pack the nuspec.

**Comment by weshaggard** [comment]
>How about I explicitly disable Pack? That way we still build the pkgproj and they don't get stale, we just don't pack the nuspec.

I'd be fine with that as well. I just want make sure we don't accidentally start shipping this package.

**Comment by ericstj** [comment]
@dotnet-bot test Innerloop OSX10.12 Release x64 Build and Test

**Pull request merged**